frontend/src/components/common/Resource/CreateButton.tsx

```diff
@@ -15,42 +15,51 @@ export default function CreateButton() {
   const { t } = useTranslation(['resource', 'frequent']);
 
-  const applyFunc = async (newItem: KubeObjectInterface) => {
-    try {
-      await apply(newItem);
-    } catch (err) {
-      let msg = t('Something went wrong…');
-      if (err instanceof Error) {
-        msg = err.message;
-      }
-      setErrorMessage(msg);
-      setOpenDialog(true);
-      throw err;
-    }
+  const applyFunc = async (newItems: KubeObjectInterface[]) => {
+    // @ts-ignore
+    await Promise.allSettled(newItems.map(newItem => apply(newItem))).then((values: any) => {
+      values.forEach((value: any, index: number) => {
+        if (value.status === 'rejected') {
+          let msg = t('Something went wrong…');
+          const kind = newItems[index].kind;
+          const name = newItems[index].metadata.name;
+          const apiVersion = newItems[index].apiVersion;
+          msg = value.reason + ` kind: ${kind} name: ${name} apiVersion: ${apiVersion}`;
+          setErrorMessage(msg);
+          setOpenDialog(true);
+          throw msg;
+        }
+      });
+    });
   };
 
-  function handleSave(newItemDef: KubeObjectInterface) {
+  function handleSave(newItemDefs: KubeObjectInterface[]) {
     const cancelUrl = location.pathname;
 
-    if (!newItemDef.metadata?.name) {
-      setErrorMessage(t('Please set a name to the resource!'));
-      return;
-    }
-
-    if (!newItemDef.kind) {
-      setErrorMessage(t('Please set a kind to the resource!'));
-      return;
+    // check if all yaml objects are valid
+    for (let i = 0; i < newItemDefs.length; i++) {
+      if (!newItemDefs[i].metadata?.name) {
+        setErrorMessage(t(`Invalid: One or more of the resource doesn't have a name property`));
+        return;
+      }
+      if (!newItemDefs[i].kind) {
+        setErrorMessage(t('Invalid: Please set a kind to the resource!'));
+        return;
+      }
     }
-
-    const newItemName = newItemDef.metadata.name;
-
+    // all resources name
+    const resourceNames = newItemDefs.map(newItemDef => newItemDef.metadata.name);
     setOpenDialog(false);
     dispatch(
-      clusterAction(() => applyFunc(newItemDef), {
-        startMessage: t('Applying {{ newItemName }}…', { newItemName }),
-        cancelledMessage: t('Cancelled applying {{ newItemName }}.', { newItemName }),
-        successMessage: t('Applied {{ newItemName }}.', { newItemName }),
-        errorMessage: t('Failed to apply {{ newItemName }}.', { newItemName }),
+      clusterAction(() => applyFunc(newItemDefs), {
+        startMessage: t('Applying {{ newItemNames }}…', { newItemNames: resourceNames.join(',') }),
+        cancelledMessage: t('Cancelled applying {{ newItemNames }}.', {
+          newItemNames: resourceNames.join(','),
+        }),
+        successMessage: t('Applied {{ newItemNames }}.', { newItemNames: resourceNames.join(',') }),
+        errorMessage: t('Failed to apply {{ newItemNames }}.', {
+          newItemNames: resourceNames.join(','),
+        }),
```

Comment:
I understand why you renamed `newItemName` but this will require the translations to be updated AFAIU. So if you change just the value as you did but keep the singular name, we don't have to update the translations.

Reply:
updated